src/reducers/trainer/training.ts

```diff
@@ -0,0 +1,51 @@
+import {trainerActionEnums} from '../../common/actionEnums/trainer';
+
+export class TrainingState {
+  public content: string;
+  public editor: HTMLTextAreaElement;
```

Comment:
I think this is wrong, we should not store UI components on the reducers, Which reasoning is behind this?